staging/src/k8s.io/component-base/zpages/statusz/statusz_test.go

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package statusz
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/blang/semver/v4"
+	"github.com/google/go-cmp/cmp"
+)
+
+var (
+	fakeStartTime               = time.Now()
+	fakeUptime                  = uptime(fakeStartTime)
+	fakeGoVersion               = "1.21"
+	fakeBinaryVersion           = semver.Version{Major: 1, Minor: 31, Patch: 0}
+	fakeEmulationVersion        = semver.Version{Major: 1, Minor: 30, Patch: 0}
+	fakeMinCompatibilityVersion = semver.Version{Major: 1, Minor: 29, Patch: 0}
+)
+
+func TestStatusz(t *testing.T) {
+	mux := http.NewServeMux()
+	reg = &fakeRegistry{}
+	Statusz{}.Install(mux)
+
+	path := "/statusz"
+	req, err := http.NewRequest("GET", fmt.Sprintf("http://example.com%s", path), nil)
+	if err != nil {
+		t.Fatalf("unexpected error while creating request: %v", err)
+	}
+
+	w := httptest.NewRecorder()
+	mux.ServeHTTP(w, req)
+
+	if w.Code != http.StatusOK {
+		t.Fatalf("want status code: %v, got: %v", http.StatusOK, w.Code)
+	}
+
+	c := w.Header().Get("Content-Type")
+	if c != "text/plain; charset=utf-8" {
+		t.Fatalf("want header: %v, got: %v", "text/plain", c)
+	}
+
+	if diff := cmp.Diff(wantResp(), string(w.Body.String())); diff != "" {
+		t.Fatalf("Unexpected diff on response (-want,+got):\n%s", diff)
+	}
+}
+
+func wantResp() string {
+	return fmt.Sprintf(
+		"\n----------------------------\ntitle: Kubernetes Statusz\ncontent_type: reference\nauto_generated: true\ndescription: Details of the status data that Kubernetes components report.\n----------------------------\n\n## Started: %v\n## Up: %s\n\n## Build Info\n--------------\n### Go version: %s\n### Binary version: %v\n### Emulation version: %v\n### Minimum Compatibility version: %v\n\n## List of useful endpoints\n--------------\nhealthz:/healthz\nlivez:/livez\nmetrics:/metrics\nreadyz:/readyz\nsli metrics:/metrics/slis\n",
```

Comment:
I think you can use actual line breaks. It's hard to see what the output is going to look like.

Reply:
I'd just store it in a var

Reply:
Done.